src/main/java/de/tum/in/www1/artemis/repository/metis/conversation/OneToOneChatRepository.java

```diff
@@ -0,0 +1,57 @@
+package de.tum.in.www1.artemis.repository.metis.conversation;
+
+import static org.springframework.data.jpa.repository.EntityGraph.EntityGraphType.LOAD;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.metis.conversation.OneToOneChat;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+@Repository
+public interface OneToOneChatRepository extends JpaRepository<OneToOneChat, Long> {
+
+    @EntityGraph(type = LOAD, attributePaths = { "conversationParticipants.user.groups" })
+    @Query("""
+                 SELECT DISTINCT oneToOneChat
+                 FROM OneToOneChat oneToOneChat
+                 LEFT JOIN oneToOneChat.conversationParticipants conversationParticipant
+                 LEFT JOIN FETCH oneToOneChat.conversationParticipants
+                 WHERE oneToOneChat.course.id = :#{#courseId}
+                 AND (oneToOneChat.lastMessageDate IS NOT NULL OR oneToOneChat.creator.id = :#{#userId})
+                 AND conversationParticipant.user.id = :#{#userId}
+                 ORDER BY oneToOneChat.lastMessageDate DESC
+            """)
+    List<OneToOneChat> findActiveOneToOneChatsOfUserWithParticipantsAndUserGroups(@Param("courseId") Long courseId, @Param("userId") Long userId);
+
+    @EntityGraph(type = LOAD, attributePaths = { "conversationParticipants.user.groups" })
+    @Query("""
+            SELECT DISTINCT  o FROM OneToOneChat o
+            LEFT JOIN FETCH o.conversationParticipants
+            WHERE
+            o.course.id = :#{#courseId}
+            AND
+            EXISTS (SELECT p FROM ConversationParticipant p WHERE p.user.id = :#{#userIdA} AND p.conversation = o)
+            AND
+            EXISTS (SELECT p FROM ConversationParticipant p WHERE p.user.id = :#{#userIdB} AND p.conversation = o)
```

Comment:
Are these sub-queries really required? A join might be better performance-wise.

Reply:
Removed subquery in 3846cca54aad0a5c67d37f7915667103fa030fe0